File: src/main/java/dev/akif/crud/simple/SimpleEntity.java
package dev.akif.crud.simple;

import dev.akif.crud.CRUDEntity;

import java.io.Serializable;

/**
 * Simple variant of {@link CRUDEntity}
 *
 * @param <I> Id type of the data
 * @param <E> Entity type, meant to be the exact type extending this class
 */
public abstract class SimpleEntity<I extends Serializable, E extends CRUDEntity<I, E>> extends CRUDEntity<I, E> {
    /**
     * Default empty constructor
     */
    public SimpleEntity() {}
}

File: api/src/main/kotlin/dev/akif/crud/simple/SimpleEntity.kt
package dev.akif.crud.simple

import dev.akif.crud.CRUDEntity
import java.io.Serializable

/**
 * Simple variant of [CRUDEntity]
 *
 * @param I Id type of the data
 */
abstract class SimpleEntity<I : Serializable> : CRUDEntity<I>()
